### DISCUSSION TITLE
Suggestions on how to handle multiple architecture applications in local repository.

### DISCUSSION THREAD

**Discussion by JasonatCU** [EUCPilots/evergreen-module#612] [Q&A]
What would be the suggested method to download, say the x86 and x64 versions of an application in a local repo?  My current method appears to be seeing two diffent apps with the same name and returning duplicate results with Get-EvergreenLibraryApp. 

For example, this is my Evergreen.json entry for 7zip.  I have entry for x64 and one for x86.  
<img width="656" alt="image" src="https://github.com/aaronparker/evergreen/assets/9887371/319ace87-5a7d-4a85-913f-57ad83158baf">

This is what my 7zip JSON looks like.  I removed the Path variable because it contains internal data.  
<img width="628" alt="image" src="https://github.com/aaronparker/evergreen/assets/9887371/a02785d2-8e00-42bd-8363-fc8d7cf5bd5b">

This is what happens with Get-EvergreenLibraryApp.  Note that it returns two version 23.01 for x86 and two for x64 .  My thought is that it sees the two 7zip entries in the Evergreen.json and returning it twice.  
<img width="750" alt="image" src="https://github.com/aaronparker/evergreen/assets/9887371/6200e0f9-f966-4690-8744-eb0d8a2de145">

My only thought would be to create a 7zip and a 7zipx86 entry in the evergreen.json, but I'd rather keep it clean and not do that if I don't have too.

**Comment by aaronparker** [comment]
7zip is returning 2 x86 exe files at the moment which could cause double ups. I'm not seeing two x64 exe installers:

```powershell
Version Architecture Type Size    Md5                              URI
------- ------------ ---- ----    ---                              ---
23.01   x86          exe  584704  58fc6de6c4e5d2fda63565d54feb9e75 https://ixpeering.dl.sourceforge.net/project/sevenzip/7-Zip/23.01/7zr.exe
23.01   x64          msi  1933312 7025502de20d7b39fb06870ab06d015b https://ixpeering.dl.sourceforge.net/project/sevenzip/7-Zip/23.01/7z2301-x64.msi
23.01   x64          exe  1589510 e5788b13546156281bf0a4b38bdd0901 https://ixpeering.dl.sourceforge.net/project/sevenzip/7-Zip/23.01/7z2301-x64.exe
23.01   ARM64        exe  1527518 3c5917f4da614ef892f055c697744b77 https://ixpeering.dl.sourceforge.net/project/sevenzip/7-Zip/23.01/7z2301-arm64.exe
23.01   ARM32        exe  1596691 d98f54ca52882376fd8b39b29628a90a https://ixpeering.dl.sourceforge.net/project/sevenzip/7-Zip/23.01/7z2301-arm.exe
23.01   x86          msi  1501184 49780e0517aa248e7fd90f50674a2645 https://ixpeering.dl.sourceforge.net/project/sevenzip/7-Zip/23.01/7z2301.msi
23.01   x86          exe  1301195 1cfb215a6fb373ac33a38b1db320c178 https://ixpeering.dl.sourceforge.net/project/sevenzip/7-Zip/23.01/7z2301.exe
```

You could have two entries in the library file that further filters the output - one each for x86 and x64
